cognite/client/_cognite_client.py

```diff
@@ -213,3 +217,64 @@ def default_oauth_interactive(
         """
         credentials = OAuthInteractive.default_for_azure_ad(tenant_id, client_id, cdf_cluster)
         return cls.default(project, cdf_cluster, credentials, client_name)
+
+    @classmethod
+    def from_dict(cls, config: dict[str, Any]) -> CogniteClient:
+        cognite_sdk_config_input = config.get("cognite")
+        if cognite_sdk_config_input is None:
+            raise ValueError("cognite section is missing in the configuration file")
+
+        global_config_input = cognite_sdk_config_input.get("global_config")
+        if global_config_input:
+            # TODO: set global config based on input
+            # GlobalConfig.from_dictionary(**global_config_input)
+            global_config = GlobalConfig()  # noqa: F841
+
+        client_config_input = cognite_sdk_config_input.get("client_config")
+        if client_config_input:
+            credentials_config_input = client_config_input.get("credentials")
+            if credentials_config_input is None:
+                raise ValueError("credentials section is missing in the configuration file")
+            elif isinstance(credentials_config_input, dict) and len(credentials_config_input) == 1:
+                credential_type, credential_provider_input = next(iter(credentials_config_input.items()))
+                credentials = CredentialProvider.from_config(
+                    credential_type=credential_type, config=credential_provider_input
+                )
+                client_config_input["credentials"] = credentials
+                client_config = ClientConfig(**client_config_input)
+            else:
+                raise ValueError("credentials section should contain exactly one key-value pair")
+        else:
+            raise ValueError("client_config section is missing in the configuration file")
+
+        return cls(client_config)
+
+    @classmethod
+    def from_yaml(cls, file_path: str | Path) -> CogniteClient:
+        # TODO: docstring, type hints, and error handling
+
+        file_path = Path(file_path)
+        if not file_path.is_file():
+            raise ValueError(f"File {file_path} is not a file")
+
+        try:
+            with file_path.open("r") as file_raw:
+                sub_template = Template(file_raw.read())  # FIXME: use string.Template or expand yaml.SafeLoader class
+
+                if not sub_template.is_valid():  # type: ignore[attr-defined]
+                    raise ValueError("Invalid template")
```

Comment:
This was added in version 3.11

Reply:
Yea 🥲 saw it after I pushed but got stuck doing some other things. Will update to only use the `substitute/safe_ substitute` that's available in 3.8>=